src/ucp/core/ucp_am.c

```diff
@@ -1162,18 +1159,23 @@ ucp_am_handler_common(ucp_worker_h worker, ucp_am_hdr_t *am_hdr, size_t hdr_size
         if (ucs_unlikely(UCS_STATUS_IS_ERR(status))) {
             goto err;
         }
-        data = desc + 1;
+        data        = desc + 1;
+        recv_flags |= UCP_AM_RECV_ATTR_FLAG_DATA;
+    } else if (am_flags & UCT_CB_PARAM_FLAG_DESC) {
+        recv_flags |= UCP_AM_RECV_ATTR_FLAG_DATA;
```

Comment:
can we do like this
if (DESC) {
  recv_flags |= UCP_AM_RECV_ATTR_FLAG_DATA;
} else if (UCP_AM_FLAG_PERSISTENT_DATA) {
  recv_desc_init(...)
}